kernel/random/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+//! A crate for generating random numbers.
+
+#![no_std]
+
+use rand_chacha::{
+    rand_core::{RngCore, SeedableRng},
+    ChaCha20Rng,
+};
+use spin::Mutex;
+
+// TODO: Ideally we'd use core::cell::LazyCell (or core::lazy::Lazy, or whatever
+// they choose to call it), but for some reason it doesn't yet allow mutable
+// access despite being !Sync.
+lazy_static::lazy_static! {
+    static ref CSPRNG: Mutex<ChaCha20Rng> = {
+        let seed = rdseed_seed()
+            .or_else(|_| rdrand_seed())
+            .unwrap_or_else(|_| {
+                log::error!("Could not generate hardware seed - using a predetermined seed. THIS IS NOT OK.");
+                [1; 32]
```

Comment:
this is okay, but it'd probably be better to explicitly initialize our RNG with a backup seed value obtained from something at least partially random, perhaps a current timer value (PIT/RTC/TSC/HPET) or something else.

For this, you can use the `spin::Once` type or OnceCell or something similar that allows you to pass in values to the init routine, instead of `lazy_static`. Might as well do that since you're initializing it from the `captain` anyway.

Reply:
Yea that's probably best. We'd need to ensure that `rand` is only initialised after the timers then.